libraries/omi/src/basic-tests.js

```diff
@@ -0,0 +1,150 @@
+      let wc = root.shadowRoot.querySelector("#wc");
+      expect(wc).to.exist;
+    });
+  });
+
+  describe("with children", function () {
+    function expectHasChildren(wc) {
+      expect(wc).to.exist;
+      let shadowRoot = wc.shadowRoot;
+      let heading = shadowRoot.querySelector("h1");
+      expect(heading).to.exist;
+      expect(heading.textContent).to.eql("Test h1");
+      let paragraph = shadowRoot.querySelector("p");
+      expect(paragraph).to.exist;
+      expect(paragraph.textContent).to.eql("Test p");
+    }
+
+    it("can display a Custom Element with children in a Shadow Root", function () {
+      this.weight = 3;
+      let root = render(<component-with-children />, scratch);
+      let wc = root.shadowRoot.querySelector("#wc");
+      expectHasChildren(wc);
+    });
+
+    it("can display a Custom Element with children in a Shadow Root and pass in Light DOM children", async function () {
+      this.weight = 3;
+      let root = render(<component-with-children-rerender />, scratch);
+
+      let wc = root.shadowRoot.querySelector("#wc");
+      await Promise.resolve();
+      root.update();
+      expectHasChildren(wc);
+      expect(wc.textContent.includes("2")).to.be.true;
+    });
+
+    it("can display a Custom Element with children in the Shadow DOM and handle hiding and showing the element", function () {
+      this.weight = 3;
+      let root = render(<component-with-different-views />, scratch);
+
+      let wc = root.shadowRoot.querySelector("#wc");
+      expectHasChildren(wc);
+      root.toggle();
+      let dummy = root.shadowRoot.querySelector("#dummy");
+      expect(dummy).to.exist;
+      expect(dummy.textContent).to.eql("Dummy view");
+      root.toggle();
+      wc = root.shadowRoot.querySelector("#wc");
+      expectHasChildren(wc);
+    });
+  });
+
+  describe("attributes and properties", function () {
+    it("will pass boolean data as either an attribute or a property", function () {
+      this.weight = 3;
+      let root = render(<component-with-properties />, scratch);
+      let wc = root.shadowRoot.querySelector("#wc");
+      let data = wc.bool || wc.hasAttribute("bool");
+      expect(data).to.be.true;
+    });
+
+    it("will pass numeric data as either an attribute or a property", function () {
+      this.weight = 3;
+      let root = render(<component-with-properties />, scratch);
+      let wc = root.shadowRoot.querySelector("#wc");
+      let data = wc.num || wc.getAttribute("num");
+      expect(parseInt(data, 10)).to.eql(42);
+    });
+
+    it("will pass string data as either an attribute or a property", function () {
+      this.weight = 3;
+      let root = render(<component-with-properties />, scratch);
+      let wc = root.shadowRoot.querySelector("#wc");
+      let data = wc.str || wc.getAttribute("str");
+      expect(data).to.eql("Omi");
+    });
+
+    it('will set boolean attributes on a Custom Element that has not already been defined and upgraded', function () {
```

Comment:
These three tests aren't being tested in the other libraries yet, and we want to run the same tests on all libraries. Please remove these three <component-with-unregistered> tests for now and update expectedResults.json (as afterwards the total will be 30 tests instead of 36)

Reply:
done